### PR TITLE
Port FunctionsAcceptance to the TCK

### PR DESCRIPTION
**Pull request by Mats-SX**

### CODE REVIEW

File: tck/features/FunctionsAcceptance.feature [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Feature: FunctionsAcceptance
+
+  Scenario: `split()`
```
Comment (petraselmer) [inline tck/features/FunctionsAcceptance.feature]: Do the backticks add anything to the scenario name?

File: tck/features/FunctionsAcceptance.feature [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+      WITH *
+      MATCH (n)
+      RETURN toInt(n.age) AS age
+      """
+    Then the result should be:
+      | age |
+      | 42  |
+    And no side effects
+
+  Scenario: `toInt()` on float
```
Comment (petraselmer) [inline tck/features/FunctionsAcceptance.feature]: Backticks?

File: tck/features/FunctionsAcceptance.feature [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+      """
+      WITH 82.9 AS weight
+      RETURN toInt(weight)
+      """
+    Then the result should be:
+      | toInt(weight) |
+      | 82            |
+    And no side effects
+
+  Scenario: `toInt()` returning `null` on non-numerical string
```
Comment (petraselmer) [inline tck/features/FunctionsAcceptance.feature]: Backticks?

File: tck/features/FunctionsAcceptance.feature [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+      """
+      WITH 'foo' AS foo_string, '' AS empty_string
+      RETURN toInt(foo_string) AS foo, toInt(empty_string) AS empty
+      """
+    Then the result should be:
+      | foo  | empty |
+      | null | null  |
+    And no side effects
+
+  Scenario: `toInt()` handling mixed number types
```
Comment (petraselmer) [inline tck/features/FunctionsAcceptance.feature]: Backticks?

File: tck/features/FunctionsAcceptance.feature [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+      """
+      WITH [2, 2.9] AS numbers
+      RETURN [n IN numbers | toInt(n)] AS int_numbers
+      """
+    Then the result should be:
+      | int_numbers |
+      | [2, 2]      |
+    And no side effects
+
+  Scenario: `toInt()` failing on `Any` type
```
Comment (petraselmer) [inline tck/features/FunctionsAcceptance.feature]: Backticks?

File: tck/features/FunctionsAcceptance.feature [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+    Given any graph
+    When executing query:
+      """
+      WITH [2, 2.9, '1.7'] AS things
+      RETURN [n IN things | toInt(n)] AS int_numbers
+      """
+    Then a SyntaxError should be raised at compile time: InvalidTypeAny
+
+  Scenario: `toInt()` on string list
```
Comment (petraselmer) [inline tck/features/FunctionsAcceptance.feature]: Are backticks needed (and same for all scenario names following this one)...

File: tck/features/FunctionsAcceptance.feature [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+
+  Scenario: `properties()` failing on string
+    Given any graph
+    When executing query:
+      """
+      RETURN properties('Cypher')
+      """
+    Then a SyntaxError should be raised at compile time: InvalidTypeString
+
+  Scenario: `properties()` failing on list of boolean
```
Comment (petraselmer) [inline tck/features/FunctionsAcceptance.feature]: -> `..failing on list of booleans`

File: tck/features/FunctionsAcceptance.feature [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+  Scenario: `toFloat()` failing on `Any` type
+    Given any graph
+    When executing query:
+      """
+      WITH [3.4, 3, '5'] AS numbers
+      RETURN [n IN numbers | toFloat(n)] AS float_numbers
+      """
+    Then a SyntaxError should be raised at compile time: InvalidTypeAny
+
+  Scenario: `toFloat()` on string list
```
Comment (petraselmer) [inline tck/features/FunctionsAcceptance.feature]: -> '..on a list of strings'

File: tck/features/FunctionsAcceptance.feature [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+  Scenario: `toString()` failing on `Any` type
+    Given any graph
+    When executing query:
+      """
+      WITH [2, 2.9, '1.7'] AS numbers
+      RETURN [n IN numbers | toString(n)] AS int_numbers
+      """
+    Then a SyntaxError should be raised at compile time: InvalidTypeAny
+
+  Scenario: `toString()` on integer list
```
Comment (petraselmer) [inline tck/features/FunctionsAcceptance.feature]: -> '..on a list of integers'

File: tck/features/FunctionsAcceptance.feature [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+    When executing query:
+      """
+      RETURN properties({name: 'Popeye', level: 9001}) AS m
+      """
+    Then the result should be:
+      | m                             |
+      | {name: 'Popeye', level: 9001} |
+    And no side effects
+
+  Scenario: `properties()` failing on integer
```
Comment (petraselmer) [inline tck/features/FunctionsAcceptance.feature]: -> 'failing on an integer literal'

File: tck/features/FunctionsAcceptance.feature [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+    Given any graph
+    When executing query:
+      """
+      RETURN properties(1)
+      """
+    Then a SyntaxError should be raised at compile time: InvalidTypeInteger
+
+  Scenario: `properties()` failing on string
```
Comment (petraselmer) [inline tck/features/FunctionsAcceptance.feature]: -> 'failing on a string literal'

**Comment by petraselmer** [comment]
@Mats-SX : I had a few re-wording changes, and there seem to be a couple of redundant backticks. However, I have edited many of my original comments on using backticks with function names. Looks good!

**Comment by Mats-SX** [comment]
@petraselmer Comments addressed, and feature file completed.

**Pull request merged**